tests/openfl/utilities/test_fed_timer.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (C) 2020-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Federation Components Timeout tests module."""
+
+import asyncio
+import os
+import pytest
+import time
+
+from openfl.utilities.fed_timer import fedtiming
+from unittest import mock
+
+
+@pytest.mark.parametrize(
+    'input,expected', [
+        ('first', 'first'),
+        (2.0, 2.0),
+        ({}, {}),
+        ([], []),
+        ('a', 'a'),
+        (True, True),
+    ])
+def test_check_sync_function_return_same_value_within_timelimit(input, expected):
+    """
+    Test that the decorated synchronous function return the output within the timeout threshold
+    Function call returns the expected output and it is asserted.
+    """
+
+    @fedtiming(timeout=3)
+    def some_sync_fn(value):
+        time.sleep(2)  # Simulate long running operation
```

Comment:
I suggest choosing the minimum possible timeout values for the tests. It's better to avoid long unit tests execution time.